borg/hashindex.pyx

```diff
@@ -149,24 +178,44 @@ cdef class NSKeyIterator:
         self.key = hashindex_next_key(self.index, <char *>self.key)
         if not self.key:
             raise StopIteration
-        cdef int *value = <int *>(self.key + self.key_size)
-        return (<char *>self.key)[:self.key_size], (_le32toh(value[0]), _le32toh(value[1]))
+        cdef uint32_t *value = <uint32_t *>(self.key + self.key_size)
+        cdef uint32_t segment = _le32toh(value[0])
+        assert <uint32_t> segment <= _MAX_VALUE, "maximum number of segments reached"
```

Comment:
is `<uint32_t>` required here (see line above)?
(maybe check this kind of construct at other places also)